src/coprocessor/dag/expr/mod.rs

```diff
@@ -379,6 +380,20 @@ impl Expression {
                 .map(Datum::F64)
                 .map(|e| Expression::new_const(e, tp))
                 .map_err(Error::from),
+            ExprType::MysqlTime => expr.get_val()
+                .decode_u64()
+                .and_then(|i| {
+                    let fsp = expr.get_field_type().get_decimal() as i8;
+                    let tp = expr.get_field_type().get_tp() as u8;
+                    let tz = if tp == types::TIMESTAMP {
+                        ctx.tz
+                    } else {
+                        FixedOffset::east(0)
```

Comment:
According to [TiDB](https://github.com/pingcap/tidb/blob/master/expression/expr_to_pb.go#L118), all timestamp should be adjust to timezone UTC.